### PR TITLE
MAP-1665 Migrate to the new Alerts API

### PR DESCRIPTION
**Pull request by tobyprivett**
### Jira link

[MAP-1665](https://dsdmoj.atlassian.net/browse/MAP-1665)

### What?

- Replaced the [Prison API](https://prison-api-dev.prison.service.justice.gov.uk/swagger-ui/index.html) (Alerts / Alert Codes / Alert Types)
with the [Alerts API](https://alerts-api-dev.hmpps.service.justice.gov.uk/swagger-ui/index.html) (Alerts / Alert Types)
- Renamed the oauth client `NomisClient` to `HmppsClient`

### Why?

The Prison API Alerts endpoint has been deprecated.

### Deployment risks  

- Sizeable refactor that may introduce as yet unknown issues
- Requires: ALERTS_API_BASE_URL 
STAGING:  https://alerts-api-dev.hmpps.service.justice.gov.uk/
PREPROD / UAT: https://alerts-api-preprod.hmpps.service.justice.gov.uk/
PROD: https://alerts-api.hmpps.service.justice.gov.uk/

[MAP-1665]: https://dsdmoj.atlassian.net/browse/MAP-1665?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app.json [hunk 1]
```diff
@@ -15,6 +15,10 @@
       "description": "Name of the storage bucket to use on AWS S3",
       "required": true
     },
+    "ALERTS_API_BASE_URL": {
+      "description": "Base URL for Alerts API",
+      "required": true
+    },
     "NOMIS_SITE_FOR_API": {
       "description": "Base URL for NOMIS API",
```
Comment (Mjwillis) [inline app.json:23]: Probably should call this prison-api

File: app.json [hunk 2]
```diff
@@ -15,6 +15,10 @@
       "description": "Name of the storage bucket to use on AWS S3",
       "required": true
     },
+    "ALERTS_API_BASE_URL": {
```
Comment (Mjwillis) [inline app.json:18]: Are all URL added to config?

File: helm_deploy/values-production.yaml [hunk 3]
```diff
@@ -41,6 +41,7 @@ generic-service:
       NOMIS_API_PATH_PREFIX: nomis_api_path_prefix_key
       NOMIS_PRISON_API_PATH_PREFIX: nomis_prison_api_path_prefix_key
       NOMIS_AUTH_PATH_PREFIX: nomis_auth_path_prefix_key
+      ALERTS_API_BASE_URL: alerts_api_base_url_key
```
Comment (Mjwillis) [inline helm_deploy/values-production.yaml:44]: Ah yes!

File: helm_deploy/values-staging.yaml [hunk 4]
```diff
@@ -36,6 +36,7 @@ generic-service:
       NOMIS_AUTH_SCHEME: nomis_auth_scheme_key
       NOMIS_API_PATH_PREFIX: nomis_api_path_prefix_key
       NOMIS_PRISON_API_PATH_PREFIX: nomis_prison_api_path_prefix_key
+      ALERTS_API_BASE_URL: alerts_api_base_url_key
```
Comment (Mjwillis) [inline helm_deploy/values-staging.yaml:39]: Where do the values get addded?

**Review by Mjwillis — APPROVED** [review]
Some comments ... but LGTM

File: app.json (hunk 1 above)
Comment (tobyprivett) [reply app.json:23]: Both Prison API and NOMIS API are used all over the app, unfortunately.

File: helm_deploy/values-staging.yaml (hunk 4 above)
Comment (tobyprivett) [reply helm_deploy/values-staging.yaml:39]: As Kube secrets

**Pull request merged**